elrond-wasm-debug/src/tx_mock/tx_async_call_data.rs

```diff
@@ -60,10 +69,32 @@ pub fn async_callback_tx_input(async_data: &AsyncCallTxData, async_result: &TxRe
         gas_limit: 1000,
         gas_price: 0,
         tx_hash: async_data.tx_hash.clone(),
+        callback_payments,
         ..Default::default()
     }
 }
 
+fn extract_callback_payments(
+    callback_contract_address: &Address,
+    async_result: &TxResult,
+    builtin_functions: &BuiltinFunctionMap,
+) -> CallbackPayments {
+    let mut callback_payments = CallbackPayments::default();
+    for async_call in &async_result.all_calls {
+        let tx_input = async_call_tx_input(async_call);
+        let token_transfers = builtin_functions.extract_token_transfers(&tx_input);
+        if &token_transfers.real_recipient == callback_contract_address {
+            if !token_transfers.is_empty() {
+                callback_payments.esdt_values = token_transfers.transfers;
+            } else {
+                callback_payments.egld_value = async_call.call_value.clone();
+            }
+            break;
+        }
+    }
+    callback_payments
+}
```

Comment:
The payment returned is the first which matches the destination address.
Given the scenario:
`A ---calls---> B ---callback(with payment X)---> A ---calls---> B ---callback(with payment Y)---> A`
Wouldn't the returned payment be `X` in both cases (instead of `X` then `Y`) since `all_calls` isn't cleared?

Reply:
TODO - will handle in a future PR

Reply:
I think this is the same behavior of the VM. We need to investigate by comparing the two.